controllers/openstackcluster_controller.go

```diff
@@ -196,24 +196,34 @@ func deleteBastion(log logr.Logger, osProviderClient *gophercloud.ProviderClient
 		return err
 	}
 
-	instance, err := computeService.GetInstanceByName(openStackCluster, fmt.Sprintf("%s-bastion", cluster.Name))
+	instanceStatus, err := computeService.GetInstanceByName(openStackCluster, fmt.Sprintf("%s-bastion", cluster.Name))
 	if err != nil {
 		return err
 	}
 
-	if instance != nil && instance.FloatingIP != "" {
-		if err = networkingService.DisassociateFloatingIP(openStackCluster, instance.FloatingIP); err != nil {
-			handleUpdateOSCError(openStackCluster, errors.Errorf("failed to disassociate floating IP: %v", err))
-			return errors.Errorf("failed to disassociate floating IP: %v", err)
+	instanceNS, err := instanceStatus.NetworkStatus()
+	if err != nil {
+		return err
+	}
```

Comment:
You should move this block inside the `instanceStatus != nil` condition, as it's only used there

Reply:
This would also have been a bug!

Reply:
nil pointer incoming :)